### PR TITLE
Add details of installing dev versions

### PR DESCRIPTION
**Pull request by hyanwong**
And add info about PEP8.

This would have been useful information for a few of my students.

### CODE REVIEW

File: docs/development.rst [hunk 1]
```diff
@@ -4,15 +4,18 @@
...
+If you want to try out the development version of ``tskit``, or add features or
+documentation improvements, please read the following. If you think there is anything
+missing, please open an `issue <http://github.com/tskit-dev/tskit/issues>`_ or
 `pull request <http://github.com/tskit-dev/tskit/pulls>`_ on GitHub!
 
 **********
 Quickstart
 **********
 
+Create a local copy of the most recent (development) version
```
Comment (jeromekelleher) [inline docs/development.rst]: This isn't really a heading --- needs to be short if it's going to work as a clickable link in the left sidebar.

File: docs/development.rst [hunk 2]
```diff
@@ -23,16 +26,30 @@ Quickstart
...
 - Make your changes in a local branch, and open a pull request on GitHub when you
   are ready. Please make sure that (a) the tests pass before you open the PR; and
   (b) your code passes PEP8 checks (see below for a git commit hook to ensure this
   happens automatically) before opening the PR.
 
+Installing your local version as the python default
+---------------------------------------------------
+
+- If you (optionally) want to install your local ``tskit`` version as the default Python
+  version, point pip to the ``python`` directory, where ``setup.py`` resides. E.g. at the
```
Comment (jeromekelleher) [inline docs/development.rst]: This is confusing to novices --- what does pointing pip at something mean? Also, what does pip -e do?

File: docs/installation.rst [hunk 3]
```diff
@@ -14,7 +14,14 @@ to other software (such as `msprime <https://github.com/tskit-dev/msprime>`_). I
...
 
-    $ python -m pip install tskit
+    $ python3 -m pip install tskit
 
-For technical reasons it is not possible to install a development version directly using the
-GitHub URL (i.e. ``pip install git+git://...`` will not work)
+********************
+Development versions
+********************
+
```
Comment (jeromekelleher) [inline docs/installation.rst]: Need some "we do not recommend this for normal use" statement.

Comment (hyanwong) [reply docs/installation.rst]: Done

File: docs/development.rst (hunk 2 above)
Comment (hyanwong) [reply docs/development.rst]: How about 

```
- If you (optionally) want to install your local ``tskit`` version as the default Python
  version, you should be able to do ``python3 -m pip install -e python`` (the ``-e`` flag
  tells pip to install from the local tskit ``python`` directory).
```

File: docs/development.rst (hunk 1 above)
Comment (hyanwong) [reply docs/development.rst]: Yes, agree. How about:

```
Getting tskit (dev)
-------------------
```

**Comment by jeromekelleher** [comment]
Can you bring this up to date please @hyanwong --- the installation docs got updated for the release.

**Comment by hyanwong** [comment]
> Can you bring this up to date please @hyanwong --- the installation docs got updated for the release.

Should be done now. Let me know if the sectioning in `development.rst` is still problematic.

**Pull request merged**